code/modules/detectivework/microscope/dnascanner.dm

```diff
@@ -51,102 +29,55 @@
+	if(..())
+		return TOPIC_HANDLED
 
 	if(href_list["scanItem"])
 		if(scanning)
 			scanning = 0
+			return TOPIC_HANDLED
 		else
-			if(bloodsamp)
-				if(closed == 1)
-					scanner_progress = 0
-					scanning = 1
-					to_chat(usr, SPAN_NOTICE("Scan initiated."))
-					update_icon()
-				else
-					to_chat(usr, SPAN_NOTICE("Please close sample lid before initiating scan."))
+			if(sample)
+				scanner_progress = 0
+				scanning = 1
+				to_chat(user, SPAN_NOTICE("Scan initiated."))
+				update_icon()
+				return TOPIC_REFRESH
 			else
-				to_chat(usr, SPAN_WARNING("Insert an item to scan."))
+				to_chat(user, SPAN_WARNING("Insert an item to scan."))
+				return TOPIC_HANDLED
 
 	if(href_list["ejectItem"])
-		if(bloodsamp)
-			bloodsamp.forceMove(src.loc)
-			bloodsamp = null
-
-	if(href_list["toggleLid"])
-		toggle_lid()
-
-	return 1
+		if(sample)
+			sample.forceMove(src.loc)
+			sample = null
+			return TOPIC_REFRESH
 
-/obj/machinery/dnaforensics/Process()
+/obj/machinery/forensic/dnascanner/Process()
 	if(scanning)
-		if(!bloodsamp || bloodsamp.loc != src)
-			bloodsamp = null
+		if(!sample || sample.loc != src)
+			sample = null
 			scanning = 0
-		else if(scanner_progress >= 100)
-			complete_scan()
 			return
-		else
-			//calculate time difference
-			var/deltaT = (world.time - last_process_worldtime) * 0.1
-			scanner_progress = min(100, scanner_progress + scanner_rate * deltaT)
-	last_process_worldtime = world.time
+		scanner_progress += scanner_rate
+		if(scanner_progress >= 100)
+			complete_scan()
 
-/obj/machinery/dnaforensics/proc/complete_scan()
-	src.visible_message(SPAN_NOTICE("\icon[src] makes an insistent chime."), 2)
+/obj/machinery/forensic/dnascanner/proc/complete_scan()
+	visible_message(SPAN_NOTICE("[src] makes an insistent chime."), 2)
+	playsound(src, 'sound/machines/chime.ogg', 40)
+	scanning = 0
+	print_report()
 	update_icon()
-	if(bloodsamp)
-		var/obj/item/paper/P = new(src)
-		P.SetName("[src] report #[++report_num]: [bloodsamp.name]")
-		P.stamped = list(/obj/item/stamp)
-		P.overlays = list("paper_stamped")
-		//dna data itself
-		var/data = "No scan information available."
-		if(bloodsamp.dna != null || bloodsamp.trace_dna != null)
-			data = "Spectometric analysis on provided sample has determined the presence of DNA.<br><br>"
-			for(var/blood in bloodsamp.dna)
-				data += "<span class='notice'>Blood type: [bloodsamp.dna[blood]]<br>DNA: [blood]</span><br><br>"
-			for(var/trace in bloodsamp.trace_dna)
-				data += "<span class='notice'>Trace DNA: [trace]</span><br><br>"
-		else
-			data += "No DNA found.<br>"
-		P.info = "<b>[src] analysis report #[report_num]</b><br>"
-		P.info += "<b>Scanned item:</b><br>[bloodsamp.name]<br>[bloodsamp.desc]<br><br>" + data
-		P.forceMove(src.loc)
-		P.update_icon()
-		scanning = 0
-		update_icon()
-	return
 
-/obj/machinery/dnaforensics/interface_interact(mob/user)
+/obj/machinery/forensic/dnascanner/interface_interact(mob/user)
```

Comment:
Duplicate definition, unless my diff is confusing me.

Reply:
axed it, was duplicate